Zemismart/__init__.py

```diff
@@ -186,9 +209,227 @@ def save_battery(self, battery):
+            return device
+        return self._device
+
+    def remove(self):
+        if self._timer_id is None:
+            raise ValueError('Unable to remove timer that wasn\'t assigned to a device')
+
+        self._verify_timer_can_be_saved(None)
+
+        if self._timer_id >= len(self._device.timers):
+            raise ValueError('Trying to remove unknown timer')
+
+        ret = self._update_timer(self._device, self._timer_id, action=1)
+
+        if ret:
+            del self._device.timers[self._timer_id]
+
+            timer_id = 0
+            for timer in self._device.timers:
+                timer._timer_id = timer_id
+                timer_id += 1
+
+            self._device = None
+            self._timer_id = None
+
+        return ret
+
+    def save(self, device=None):
+        self._verify_timer_can_be_saved(device)
+        if len(self._select_device(device).timers) >= 4:
+            raise ValueError('You can have only 4 timers')
+
+        timer_id = self._timer_id
+
+        if device is not None:
+            timer_id = len(device.timers)
+        elif timer_id is None:
+            timer_id = len(self._device.timers)
+        elif timer_id >= len(self._device.timers):
+            raise ValueError('Trying to update unknown timer')
+
+        ret = self._update_timer(self._select_device(device), timer_id, self._enabled, self._target_position,
+                                 self._repeats, self._hours, self._minutes)
+
+        if ret:
+            self._device = self._select_device(device)
+            self._timer_id = timer_id
+
+            if len(self._device.timers) - 1 >= timer_id:
+                self._device.timers[timer_id] = self
+            else:
+                self._device.timers.append(self)
+
+        return ret
+
+    def disable(self):
+        self._enabled = False
+
+    def enable(self):
+        self._enabled = True
+
+    def set_repeats(self, repeats):
+        self._repeats = repeats
+
+    def set_hours(self, hours):
+        if not 0 <= hours <= 23:
+            raise ValueError('The value must be between 0 and 23')
+
+        self._hours = hours
+
+    def set_minutes(self, minutes):
+        if not 0 <= minutes <= 59:
+            raise ValueError('The value must be between 0 and 59')
+
+        self._minutes = minutes
+
+    def __repr__(self):
+        ret = ''
+
+        if self._device is None:
+            ret += 'Unassigned'
+        else:
+            ret += 'Assigned'
+
+        ret += ' '
+
+        if self._enabled:
+            ret += 'active'
+        else:
+            ret += 'inactive'
+
+        ret += ' timer'
+
+        if self._timer_id is None:
+            ret += ' w/o id'
+        else:
+            ret += '#%d' % self._timer_id
+
+        ret += ': set to %s%% at %02d:%02d ' % (self._target_position, self._hours, self._minutes)
+
+        if self._repeats:
+            ret += 'on '
+            repeats = []
+            if self._repeats == self.REPEAT_EVERY_DAY:

```

Comment:
With this implementation, only the first repeat day will be reported (unless it repeats everyday), since you are using `elif`

I suggest doing something like below: 
```suggestion
            if self._repeats == self.REPEAT_EVERY_DAY:
                repeats.append('everyday')
            else:
                if self._repeats & self.REPEAT_MONDAY:
                    repeats.append('Monday')
                if self._repeats & self.REPEAT_TUESDAY:
                    repeats.append('Tuesday')
                if self._repeats & self.REPEAT_WEDNESDAY:
                    repeats.append('Wednesday')
                if self._repeats & self.REPEAT_THURSDAY:
                    repeats.append('Thursday')
                if self._repeats & self.REPEAT_FRIDAY:
                    repeats.append('Friday')
                if self._repeats & self.REPEAT_SATURDAY:
                    repeats.append('Saturday')
                if self._repeats & self.REPEAT_SUNDAY:
                    repeats.append('Sunday')
```

Reply:
OMG 🤦‍♂️ As you could've guessed — I didn't test it well. I have only "every day" timers, so I completely missed the custom schedules. Thanks!